src/audio/pipeline/pipeline-graph.c

```diff
@@ -358,18 +358,33 @@ int pipeline_reset(struct pipeline *p, struct comp_dev *host)
 int pipeline_for_each_comp(struct comp_dev *current,
 			   struct pipeline_walk_context *ctx, int dir)
 {
-	struct list_item *buffer_list = comp_buffer_list(current, dir);
+	int flag = dir & 0x80000000;
+	struct list_item *buffer_list;
 	struct list_item *clist;
 	struct comp_buffer *buffer;
 	struct comp_dev *buffer_comp;
+	static unsigned int common_dbg;
+	int reason = 0;
+
+	dir &= ~0x80000000;
+	buffer_list = comp_buffer_list(current, dir);
 
 	/* run this operation further */
 	list_for_item(clist, buffer_list) {
+		reason = 3;
```

Comment:
use defines for 'reason' values, it's just unreadable for the rest of us otherwise.